controlplane/pkg/nsm/nsm.go

```diff
@@ -107,7 +108,20 @@ func (srv *networkServiceManager) request(ctx context.Context, request nsm.NSMRe
 
 	// 3. get dataplane
 	srv.serviceRegistry.WaitForDataplaneAvailable(srv.model, DataplaneTimeout)
-	dp, err := srv.model.SelectDataplane()
+	dp, err := srv.model.SelectDataplane(func(dp *model.Dataplane) bool {
+		if request.IsRemote() {
+			return findRemoteMechanism(dp.RemoteMechanisms, remote_connection.MechanismType_VXLAN) != nil
```

Comment:
Maybe have the selector func as a separate, not inline.

Reply:
@nickolaev because VxLAN will be selected as remote mechanism at the next step:
https://github.com/networkservicemesh/networkservicemesh/blob/ed73f9acab5a733a0c833ca29849146b53099ba3/controlplane/pkg/nsm/nsm_mechanims.go#L55-L57

Reply:
Function is inline because I'm using closures to get arguments of request method.